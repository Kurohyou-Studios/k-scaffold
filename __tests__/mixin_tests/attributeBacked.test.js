import * as fs from 'fs/promises';

import { describe,it,expect } from 'vitest';

import { k } from '../testFramework';

const htmlURL = new URL('../../test_sheet/build/attributeBacked.test.html',import.meta.url);

describe('attributeBacked entry mixins',()=>{
  it('Should create the attributeBacked mixins',async ()=>{
    const html = await fs.readFile(htmlURL,'utf8');
    expect(html).toMatchSnapshot();
  })
})