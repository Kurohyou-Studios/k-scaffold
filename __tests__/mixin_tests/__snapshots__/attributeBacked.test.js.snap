// Vitest Snapshot v1

exports[`attributeBacked entry mixins > Should create the attributeBacked mixins 1`] = `
"<img name=\\"attr_image\\" src=\\"https://dne.com\\" data-i18n-alt=\\"image\\" title=\\"@{image}\\"/><span name=\\"attr_span\\" title=\\"@{span}\\"></span>
<div name=\\"attr_div\\" title=\\"@{div}\\">
</div>
<fieldset class=\\"repeating_basic-fieldset\\"><img name=\\"attr_image\\" src=\\"https://dne.com\\" data-i18n-alt=\\"image\\" title=\\"@{repeating_basic-fieldset_$X_image}\\"/><span name=\\"attr_span\\" title=\\"@{repeating_basic-fieldset_$X_span}\\"></span>
  <div name=\\"attr_div\\" title=\\"@{repeating_basic-fieldset_$X_div}\\">
  </div>
</fieldset><span class=\\"add-a-class\\" hidden=\\"\\"></span>
<fieldset class=\\"repeating_class-fieldset\\"><img name=\\"attr_image\\" src=\\"https://dne.com\\" data-i18n-alt=\\"image\\" title=\\"@{repeating_class-fieldset_$X_image}\\"/><span name=\\"attr_span\\" title=\\"@{repeating_class-fieldset_$X_span}\\"></span>
  <div name=\\"attr_div\\" title=\\"@{repeating_class-fieldset_$X_div}\\">
  </div>
</fieldset>
<button class=\\"repcontrol-button repcontrol-button--add\\" name=\\"act_add-custom-fieldset\\" type=\\"action\\" title=\\"%{add-custom-fieldset}\\">
</button>
<button class=\\"repcontrol-button repcontrol-button--edit\\" name=\\"act_edit-custom-fieldset\\" type=\\"action\\" title=\\"%{edit-custom-fieldset}\\">
</button>
<fieldset class=\\"repeating_custom-fieldset\\"><img name=\\"attr_image\\" src=\\"https://dne.com\\" data-i18n-alt=\\"image\\" title=\\"@{repeating_custom-fieldset_$X_image}\\"/><span name=\\"attr_span\\" title=\\"@{repeating_custom-fieldset_$X_span}\\"></span>
  <div name=\\"attr_div\\" title=\\"@{repeating_custom-fieldset_$X_div}\\">
  </div>
</fieldset><span class=\\"inline-fieldset\\" hidden=\\"\\"></span>
<fieldset class=\\"repeating_inline-fieldset\\">
  <input class=\\"display-control\\" name=\\"attr_display_state\\" value=\\"short-display\\" hidden=\\"\\" type=\\"radio\\" title=\\"@{repeating_inline-fieldset_$X_display_state}\\"/>
  <div class=\\"inline-fieldset__summary display-target\\">
    <label class=\\"pointer\\">
      <input name=\\"attr_collapse\\" value=\\"1\\" hidden=\\"\\" type=\\"checkbox\\" title=\\"@{repeating_inline-fieldset_$X_collapse}\\"/><span class=\\"inline-fieldset__summary__text\\" name=\\"attr_name\\" title=\\"@{repeating_inline-fieldset_$X_name}\\"></span>
    </label>
  </div>
  <input class=\\"display-control\\" name=\\"attr_display_state\\" value=\\"display\\" checked=\\"\\" hidden=\\"\\" type=\\"radio\\" title=\\"@{repeating_inline-fieldset_$X_display_state}\\"/>
  <div class=\\"inline-fieldset__detail display-target\\">
    <input class=\\"collapse\\" name=\\"attr_collapse\\" value=\\"1\\" type=\\"checkbox\\" title=\\"@{repeating_inline-fieldset_$X_collapse}\\"/><img name=\\"attr_image\\" src=\\"https://dne.com\\" data-i18n-alt=\\"image\\" title=\\"@{repeating_inline-fieldset_$X_image}\\"/><span name=\\"attr_span\\" title=\\"@{repeating_inline-fieldset_$X_span}\\"></span>
    <div name=\\"attr_div\\" title=\\"@{repeating_inline-fieldset_$X_div}\\">
    </div>
  </div>
</fieldset>
<button class=\\"repcontrol-button repcontrol-button--add repcontrol-button--inline\\" name=\\"act_add-inline-fieldset\\" type=\\"action\\" title=\\"%{add-inline-fieldset}\\">
</button>
<button class=\\"repcontrol-button repcontrol-button--edit repcontrol-button--inline\\" name=\\"act_edit-inline-fieldset\\" type=\\"action\\" title=\\"%{edit-inline-fieldset}\\">
</button>"
`;
